src/features/navigation/__tests__/useInitialScreenConfig.web.test.tsx

```diff
@@ -0,0 +1,100 @@
+
+const mockSettings = {
+  enableNativeCulturalSurvey: false,
+}
+
+jest.mock('features/auth/settings', () => ({
+  useAppSettings: jest.fn(() => ({
+    data: mockSettings,
+  })),
+}))
+
+describe('useInitialScreen()', () => {
+  afterAll(async () => {
+    await storage.clear('has_seen_tutorials')
+    await storage.clear('has_seen_eligible_card')
+  })
+
+  it.each`
+    hasSeenTutorials | hasSeenEligibleCard | isLogged | userProfile                                                                                  | isNativeCulturalSurveyActive | expectedScreen                    | expectedAnalyticsScreen
+    ${true}          | ${true}             | ${true}  | ${{ needsToFillCulturalSurvey: false, showEligibleCard: false, recreditAmountToShow: null }} | ${true}                      | ${'TabNavigator'}                 | ${'Home'}
+    ${null}          | ${true}             | ${true}  | ${{ needsToFillCulturalSurvey: true, showEligibleCard: false, recreditAmountToShow: null }}  | ${false}                     | ${'CulturalSurvey'}               | ${'CulturalSurvey'}
+    ${true}          | ${true}             | ${true}  | ${{ needsToFillCulturalSurvey: true, showEligibleCard: false, recreditAmountToShow: null }}  | ${false}                     | ${'CulturalSurvey'}               | ${'CulturalSurvey'}
+    ${true}          | ${true}             | ${true}  | ${{ needsToFillCulturalSurvey: true, showEligibleCard: true, recreditAmountToShow: null }}   | ${false}                     | ${'CulturalSurvey'}               | ${'CulturalSurvey'}
+    ${true}          | ${true}             | ${true}  | ${{ needsToFillCulturalSurvey: true, showEligibleCard: true, recreditAmountToShow: null }}   | ${true}                      | ${'CulturalSurveyIntro'}          | ${'CulturalSurveyIntro'}
+    ${true}          | ${null}             | ${true}  | ${{ needsToFillCulturalSurvey: true, showEligibleCard: true, recreditAmountToShow: null }}   | ${true}                      | ${'EighteenBirthday'}             | ${'EighteenBirthday'}
+    ${true}          | ${null}             | ${true}  | ${{ needsToFillCulturalSurvey: true, showEligibleCard: true, recreditAmountToShow: 3000 }}   | ${true}                      | ${'RecreditBirthdayNotification'} | ${'RecreditBirthdayNotification'}
+    ${true}          | ${true}             | ${false} | ${{ needsToFillCulturalSurvey: true, showEligibleCard: false, recreditAmountToShow: null }}  | ${true}                      | ${'TabNavigator'}                 | ${'Home'}
+    ${true}          | ${true}             | ${false} | ${{ needsToFillCulturalSurvey: false, showEligibleCard: true, recreditAmountToShow: null }}  | ${true}                      | ${'TabNavigator'}                 | ${'Home'}
+    ${null}          | ${true}             | ${false} | ${{ needsToFillCulturalSurvey: false, showEligibleCard: false, recreditAmountToShow: null }} | ${true}                      | ${'TabNavigator'}                 | ${'Home'}
```

Comment:
est-ce que c'est utile de tester tous les cas ? On pourrait pas juste tester ceux propres au web ? (btw ça aurait pu être ajouté dans un nouveau commit mais c'est du détail 🤓)

Reply:
**si on veut tester tout ça**, dans ce cas on ne devrait pas faire un seul `it.each`

il y a plusieurs règles qui sont testées en meme temps (ça se voit facilement, il y a valeur `expected` dans les inputs du test paramétré)

on devrait faire plusieurs tests, un par règle testé

Reply:
Effectivement c'est pas utile de tester les autres cas